### ISSUE TITLE
Using dependency injection with Assetic

### ISSUE THREAD

**Issue by leperse** [symfony/assetic-bundle#448]
Once I upgraded my application to Symfony 3.4 I get these deprecations : 

`User Deprecated: The "assetic.filter_manager" service is private, getting it from the container is deprecated since Symfony 3.2 and will fail in 4.0. You should either make the service public, or stop using the container directly and use dependency injection instead.`

`User Deprecated: The "assetic.filter.cssrewrite" service is private, getting it from the container is deprecated since Symfony 3.2 and will fail in 4.0. You should either make the service public, or stop using the container directly and use dependency injection instead.`

How could I fix that and use dependency injection instead.

**Comment by smcjones** [comment]
This can probably be fixed by removing "public: true" on services that are used throughout the bundle. Are there downsides to this approach?

**Comment by PierreRambaud** [comment]
:+1:  same problem here

**Comment by mhnrm** [comment]
+1

**Comment by vess00** [comment]
My solution to this issue, when updating to 3.4, was to modify services.yml by adding the following under the "services" section:

`Symfony\Bundle\AsseticBundle\AsseticBundle:
        public: true`

Your services.yml will look something like this:
`services:
    Symfony\Bundle\AsseticBundle\AsseticBundle:
        public: true` 

By doing this all dependency warnings go away. This is how you can fix it. How to use dependency injection - still trying to figure it out. 
Hope this helps.

**Comment by Payfu** [comment]
**vess00** : Sorry it's not clear for me, could you explain more ?

**Comment by vess00** [comment]
Hi Payfu, sure - I just re-edited my post because everything was in a single line.  
In your "config.yml", you need to add under "services:" what I have written above. Then when you run the site and all deprecated warnings should be gone. Let me know if that works for you. Best.

**Comment by Payfu** [comment]
I understand now where my problem was. 

At first you wrote "_Your **services.yml** will look something like this:_" but you meant "**config.yml**" !

So my config.yml is like this :

```
services:
    Symfony\Bundle\AsseticBundle\AsseticBundle:
        public: true
```

Anyway, thank you for your help. 👍

**Comment by vess00** [comment]
Yep - I just edited my previous post to change "services.yml" to "config.yml" ... It was late when I wrote it :) ...

**Comment by mpomet** [comment]
@vess00 I just tried your workaround but it doesn't seems to work. In fact, the very first time assetic is used, the deprecation warning shows up. Then on reloading, the warning disappears.
I think if you clean your cache you will see that it doesn't work.

Do someone have another solution ?

**Comment by smcjones** [comment]
My workaround was to abandon this repository in its entirety and use gulp. While it is very well designed, it is outdated. There is no reason other than extensive legacy use to rely on this repository anymore.

**Comment by stof** [comment]
@smcjones I totally agree with you, and this is the main reason I haven't worked on this bundle since a while (and probably the same reason for other members of the core team). I'm not using AsseticBundle since quite some time.

**Comment by vess00** [comment]
@mpomet you are probably right. I was on a journey to update a Symfony 3.3 project to 4.0. And when going through 3.4 I ran into these warnings. Now, I have updated to 4.0 and have completely replaced Assetic with Webpack Encore. @smcjones, @stof - I think this is the better fit for all new Symfony 4 projects. What do you think guys?

**Comment by stof** [comment]
@vess00 using a modern frontend tooling (be it webpack with Encore, or just Gulp for people not being that modern yet to use modules everywhere) is indeed a better fit (as it allows frontend devs to use tools built for them by their own community). And this is precisely the reason why the maintenance of Assetic and AsseticBundle is in a bad state currently.
